.github/workflows/test.yml

```diff
@@ -30,6 +30,9 @@ jobs:
             ~/.gradle/wrapper
           key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
 
+      - name: Compile kotlin
+        run: ./gradlew compileTestKotlin --parallel
```

Comment:
When I tried making CI faster the other day doing this I recall it turned out to significantly increase the time it took to do the initial build job without a clear speed gain on the test run jobs, making everything overall slower. 🤷🏻‍♀️